src/multitaper.jl

```diff
@@ -0,0 +1,518 @@
+        mt_pgram!(destination[:, time_index], subepoch, config.mt_config)
+    end
+    return Spectrogram(destination, config.mt_config.freq, config.time)
+end
+
+allocate_output(config::MTSpectrogramConfig{T}) where {T} = Matrix{fftabs2type(T)}(undef, config.mt_config.n_freq_bins, config.n_time_points)
+
+"""
+    mt_spectrogram(signal::AbstractVector{T}, n::Int=length(s) >> 3,
+                                  n_overlap::Int=n >> 1; fs::Int=1,
+                                  onesided::Bool=T <: Real, kwargs...) where {T} -> Spectrogram
+
+Compute a multitaper spectrogram. Any additional keyword arguments accepted by [`MTConfig`](@ref)
+may be passed to configure the tapering. See also [`mt_spectrogram!`](@ref).
+"""
+function mt_spectrogram(signal::AbstractVector{T}, n::Int=length(signal) >> 3,
+                                  n_overlap::Int=n >> 1; fs::Int=1,
+                                  onesided::Bool=T <: Real, kwargs...) where {T}
+    config = MTSpectrogramConfig{T}(length(signal), n, n_overlap; fs=fs,
+                                              onesided=onesided, fft_flags = FFTW.ESTIMATE, kwargs...)
+    X = allocate_output(config)
+    return mt_spectrogram!(X, signal, config)
+end
+
+#####
+##### Multitapered cross spectral matrix
+#####
+
+struct CrossSpectral{T, F}
+    values::T
+    freq::F
+end
+
+freq(c::CrossSpectral) = c.freq
+
+struct MTCrossSpectraConfig{T,T1,T2,T3,T4,F,T5,T6,C<:MTConfig{T}}
+    n_channels::Int
+    weighted_evals::T1
+    x_mt::T2
+    demean::Bool
+    mean_per_channel::T3
+    demeaned_signal::T4
+    freq::F
+    freq_range::T5
+    freq_inds::T6
+    mt_config::C
+end
+
+"""
+    MTCrossSpectraConfig{T}(n_channels, n_samples; fs=1, demean=true, low_bias=true, freq_range=nothing, nw=4, kwargs...) where T
+
+Creates a configuration object used for [`mt_cross_spectral!`](@ref) as well as [`mt_coherence!`](@ref).
+
+
+* `n_samples`: the number of samples to be used as input when computing multitaper periodograms
+    with this configuration. Used for pre-allocating temporary buffers.
+* `fs`: the number of samples per second of the input signal
+* `demean`: if `true`, the channelwise mean will be subtracted from the input signals before the cross spectral values are computed.
+* `low_bias`: if `true`, keeps only tapers with eigenvalues > `0.9`.
+* `freq_range`: if `nothing`, all frequencies are retained. Otherwise, only frequencies between `first(freq_range)` and `last(freq_range)` are retained.
+
+Any keywords accepted by [`MTConfig`](@ref) may be passed here.
+"""
+function MTCrossSpectraConfig{T}(n_channels, n_samples; fs=1, demean=true, low_bias=true, freq_range=nothing, nw=4, ntapers = 2 * nw - 1, kwargs...) where T
+    if demean
+        mean_per_channel = Vector{T}(undef, n_channels)
+        demeaned_signal = Matrix{T}(undef, n_channels, n_samples)
+    else
+        mean_per_channel = nothing
+        demeaned_signal = nothing
+    end
```

Comment:
I don't know how standard demeaning the input to a csd is, but MNE does it and we wanted to match their coherences, so I wanted the option to do it here. Of course, you could just ask the user to do it, but since we have all the structure here anyway to do things in-place and allocation-free we might as well offer it for this.